src/d3a/models/strategy/storage.py

```diff
@@ -276,7 +270,10 @@ def event_bid_traded(self, *, market_id, bid_trade):
 
     def event_market_cycle(self):
         super().event_market_cycle()
-        self.update_market_cycle_offers(self.break_even[self.area.now].sell)
+
+        self.offer_update.update_market_cycle_offers(self)
+        for market in self.area.all_markets[:-1]:
```

Comment:
Isn't this the same as past_markets ? Moreover, won't this fail in case we have more than one future market?

Reply:
Not really, all_markets returns the list of all active_markets, whereas past_markets as the name says returns the list of all closed past_markets.

And sorry,  I didn't get it, why would it fail if we have more than one future markets. We already had an integration_test to ensure to have the same results in both cases(either one future market OR multiple future_markets).

Let me know if I haven't understood you correctly or making any mistake in terms of understanding.

Reply:
Yeah, its weird that the integrationtest does not fail. I also would vote for removing the `[:-1]` and reset the counter for all active markets.

Reply:
The reason I omit the most recent active_market here is there is no bid placed in this market. So there is no reason to update it here. Let me know if its clear.

Reply:
Yes! I got it wrong. You are right. It doesnt make sense to set the counter in the most recent market.

Reply:
@spyrostz Exactly!